contracts/rdt/RevenueDistributionToken.sol

```diff
@@ -0,0 +1,414 @@
+    {
+        require(receiver_ != address(0), "RDT:D:ZERO_RECEIVER");
+        _claim(receiver_);
+        _mint(shares_ = asset_, asset_, receiver_, msg.sender);
+    }
+
+    function depositWithPermit(
+        uint256 asset_,
+        address receiver_,
+        uint256 deadline_,
+        uint8 v_,
+        bytes32 r_,
+        bytes32 s_
+    ) external virtual override nonReentrant returns (uint256 shares_) {
+        require(receiver_ != address(0), "RDT:DWP:ZERO_RECEIVER");
+        _claim(receiver_);
+        ERC20(asset).permit(msg.sender, address(this), asset_, deadline_, v_, r_, s_);
+        _mint(shares_ = asset_, asset_, receiver_, msg.sender);
+    }
+
+    function mint(uint256 shares_, address receiver_) external virtual override nonReentrant returns (uint256 assets_) {
+        _mint(shares_, assets_ = shares_, receiver_, msg.sender);
+    }
+
+    function mintWithPermit(
+        uint256 shares_,
+        address receiver_,
+        uint256 maxAssets_,
+        uint256 deadline_,
+        uint8 v_,
+        bytes32 r_,
+        bytes32 s_
+    ) external virtual override nonReentrant returns (uint256 assets_) {
+        require((assets_ = previewMint(shares_)) <= maxAssets_, "RDT:MWP:INSUFFICIENT_PERMIT");
+
+        ERC20(asset).permit(msg.sender, address(this), maxAssets_, deadline_, v_, r_, s_);
+        _mint(shares_, assets_, receiver_, msg.sender);
+    }
+
+    function withdraw(
+        uint256 asset_,
+        address receiver_,
+        address owner_
+    ) external virtual override nonReentrant returns (uint256 shares_) {
+        _claim(receiver_);
+        _burn(shares_ = asset_, asset_, receiver_, owner_, msg.sender);
+    }
+
+    function redeem(
+        uint256 shares_,
+        address receiver_,
+        address owner_
+    ) external override returns (uint256 assets_) {
+        _claim(receiver_);
+        _burn(shares_, assets_ = shares_, receiver_, owner_, msg.sender);
+    }
+
+    function claim(address receiver_) external virtual nonReentrant returns (uint256 claimableRevenueAssets_) {
+        claimableRevenueAssets_ = _claim(receiver_);
+    }
+
+    function migrate(
+        address receiver_,
+        address xTRI_,
+        uint256 xTRIAmount_
+    ) external virtual nonReentrant returns (uint256 shares_) {
+        require(ERC20(xTRI_).transferFrom(receiver_, address(this), xTRIAmount_), "RDT:M:INSUFFICIENT_PERMIT");
+        uint256 triBalanceBefore_ = ERC20(asset).balanceOf(address(this));
+        ITriBar(xTRI_).leave(xTRIAmount_);
+        uint256 triBalanceUnstaked_ = ERC20(asset).balanceOf(address(this)) - triBalanceBefore_;
+
+        _mint(shares_ = triBalanceUnstaked_, triBalanceUnstaked_, receiver_, address(this));
+
+        emit Migrated(receiver_, xTRI_, triBalanceUnstaked_, xTRIAmount_);
+    }
+
+    /**************************/
+    /*** Internal Functions ***/
+    /**************************/
+
+    function _mint(
+        uint256 shares_,
```

Comment:
since `shares_` and `triAmount` will always be same, why do we have this extra input variable?